src/server.rs

```diff
@@ -904,6 +913,133 @@ impl WalletService for Arc<crate::wallet::Wallet> {
+                            block_id,
+                            confirmation_time,
+                        }) => Some(Confirmation {
+                            height: block_id.height,
+                            block_hash: Some(ReverseHex::encode(&block_id.hash)),
+                            timestamp: Some(Timestamp {
+                                seconds: confirmation_time as i64,
+                                nanos: 0,
+                            }),
+                        }),
+                        ChainPosition::Unconfirmed(last_seen) => Some(Confirmation {
+                            height: 0,
+                            block_hash: None,
+                            timestamp: Some(Timestamp {
+                                seconds: last_seen as i64,
+                                nanos: 0,
+                            }),
+                        }),
+                    },
+                })
+                .collect(),
+        };
+        Ok(tonic::Response::new(response))
+    }
+
+    async fn send_transaction(
+        &self,
+        request: tonic::Request<SendTransactionRequest>,
+    ) -> Result<tonic::Response<SendTransactionResponse>, tonic::Status> {
+        let SendTransactionRequest {
+            destinations,
+            fee_rate,
+            op_return_message,
+        } = request.into_inner();
+
+        // Extract fee rate per vbyte
+        let fee_rate_per_vbyte = match fee_rate {
+            Some(fee_rate) => match fee_rate.fee {
+                Some(send_transaction_request::fee_rate::Fee::SatPerVbyte(rate)) => {
+                    Some(Amount::from_sat(rate))
+                }
+                _ => None,
+            },
+            None => None,
+        };
+
+        // Or hard-coded fee
+        let fee = match fee_rate {
+            Some(fee_rate) => match fee_rate.fee {
+                Some(send_transaction_request::fee_rate::Fee::Sats(sats)) => {
+                    Some(Amount::from_sat(sats))
+                }
+                _ => None,
+            },
+            None => None,
+        };
+
+        // Parse and validate all destination addresses, but assume network valid
+        let destinations_validated = destinations
+            .iter()
+            .map(|(address, amount)| {
+                bdk_wallet::bitcoin::Address::from_str(address)
+                    .map_err(|e| {
+                        tonic::Status::invalid_argument(format!(
+                            "could not parse bitcoin address: {}",
+                            e
+                        ))
+                    })
+                    .map(|addr| (addr.assume_checked(), *amount))
+            })
+            .collect::<Result<HashMap<bdk_wallet::bitcoin::Address, u64>, tonic::Status>>()?;
+
+        let txid = self
+            .send_wallet_transaction(
+                destinations_validated,
+                fee_rate_per_vbyte,
```

Comment:
If  `fee_rate_per_vbyte` and `fee` should not be set simultaneously, then this should be captured via the type system